### DISCUSSION TITLE
Slowing Down

### DISCUSSION THREAD

**Discussion by notgull** [smol-rs/smol#267] [General]
Hello,

As of Monday I will be starting a new full time job. This means that I won't have as much time to dedicate to open source software as I have been over the past year. Therefore I will not be actively adding new features to `smol-rs` crates. I will still do bugfixes, respond to issues/PRs, and other general maintenance items. But, once smol 2.0 is released, I will not add anything else to these crates. Which is probably good, since the crates are pretty much complete.

Here's some highlights from the past year:

- [Reimplementing the Windows backend for `polling` in pure Rust](https://github.com/smol-rs/polling/pull/88), removing our only C dependency
- [Reworking `event-listener` to be much faster and `no_std`](https://github.com/smol-rs/event-listener/pull/51)
- Adding [metadata](https://github.com/smol-rs/async-task/pull/33) and [panic propagation](https://github.com/smol-rs/async-task/pull/37) to `async-task`.
- [This tracker for potentially useful features](https://github.com/smol-rs/smol/issues/244)
- Creating/extracting the [`piper`](https://github.com/smol-rs/piper) and [`async-signal`](https://github.com/smol-rs/async-signal) crates.
- And more!

Things that I meant to do that I never got around to, that would probably be much appreciated if someone wanted to take them up:

- [`async-process`](https://github.com/smol-rs/async-process) uses signals to handle process events. This is inefficient and doesn't scale to high numbers of processes. A more optimized implementation would use `pidfd` on Linux, `EVFILT_PROC` on BSD, and waitable handles on Windows.
- Speaking of [`async-process`](https://github.com/smol-rs/async-process), it doesn't necessarily need to spawn another entire thread to run the process reactor. It can very easily be made `async` and run as a task. This would likely eliminate some overhead.
- Speaking of waitable handles, I *kind of* solved https://github.com/smol-rs/async-io/issues/25 but not really. Right now the waitable handles just use a thread pool, but it could be vastly more efficient with `NtAssociateWaitCompletionPacket`.
- Speaking of that, I never exposed waitable handles in `async-io`.
- The "next task" optimization for `async-executor` is low hanging fruit that would vastly improve our microbenchmarks.

It's been a pleasure. I'm not going anywhere, I just won't be opening as many PRs... although that will probably be a relief to not get spammed by me for reviews :-)

- John "notgull" Nunley

**Comment by taiki-e** [comment]
Thanks for all your hard work!
